src/settings-api.php

```diff
@@ -1,16 +1,68 @@
 <?php
 
-function wp_parsely_settings_api_init() {
-	register_rest_route( 'wp-parsely/v1', 'settings', array(
-		'methods' => 'GET',
-		'callback' => 'wp_parsely_settings_http_get',
-	) );
+/**
+ * Check if a given request has access to get items
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return bool
+ */
+function wp_parsely_settings_api_permissions_check( WP_REST_Request $request ) {
+	return current_user_can( Parsely::CAPABILITY );
 }
-add_action( 'rest_api_init', 'wp_parsely_settings_api_init' );
 
-function wp_parsely_settings_http_get() {
+/**
+ * Getter
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return WP_REST_Response
+ */
+function wp_parsely_settings_api_read_settings( WP_REST_Request $request ) {
 	global $parsely;
-	$settings = $parsely->get_options();
-	unset( $settings['api_secret'] );
-	return $settings;
+	return new WP_REST_Response( $parsely->get_options(), 200 );
 }
+
+/**
+ * Setter
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return WP_REST_Response
+ */
+function wp_parsely_settings_api_write_settings( WP_REST_Request $request ) {
+	global $parsely, $wp_settings_errors;
+
+	// `$parsely->validate_options` uses `add_settings_error`
+	// TODO: Decouple!
+	require_once ABSPATH . 'wp-admin/includes/template.php';
```

Comment:
This is pretty gross.  We should refactor the validation function:
https://github.com/Parsely/wp-parsely/blob/e1c3e43d98ce7e45d4c990421a690a677abdf829/src/class-parsely.php#L620
...to not call the Settings API functionality.

Let's do it in a follow-up, though, so this unblocks development on the updating side of things.